src/group_theory/perm/support.lean

```diff
@@ -0,0 +1,530 @@
+/-
+Copyright (c) 2018 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes, Aaron Anderson, Yakov Pechersky
+-/
+import data.set.finite
+import group_theory.perm.basic
+import data.list.perm
+import dynamics.fixed_points.basic
+
+/-!
+# Support of a permutation
+
+## Main definitions
+
+In the following, `f g : equiv.perm α`.
+
+* `equiv.perm.involutive`: `f.involutive` when it is its own inverse, and so are powers of it.
+* `equiv.perm.support`: the elements `x : α` that are not fixed by `f`.
+* `equiv.perm.disjoint`: two permutations `f` and `g` are `disjoint` iff their `support`
+  are disjoint.
+
+## Implementation details
+
+A perm's `support` need not necessarily be finite. To facilitate this, `f.support` is provided
+as a `set` rather than a `finset`. However, often we want to deal with the cardinality of a
+`support`. Since one can have finite support even if the underlying type if not `fintype`,
+instead of juggling `[fintype ↥(f.support)]`, we use instead `set.finite f.support`. That leads
+to lemma statements that mention `(h : set.finite f.support).to_finset.card`, and sometimes
+`letI : fintype ↥(f.support) := h.fintype` in proofs.
+
+-/
+
+open equiv set
+
+namespace equiv.perm
+
+variables {α : Type*}
+
+section involutive
+
+-- TODO: should this be refactored as a prop on any group element,
+-- with application refactored to be about mul_action?
+
+variable (f : perm α)
+
+/-- A permutation `f : perm α` is `involutive` if its inverse is equal to itself. -/
+def involutive (f : perm α) : Prop := f⁻¹ = f
```

Comment:
What is the reason for including `involutive` in this file?

Reply:
I thought it would be useful for some of the proofs about `sign` or `is_cycle` but haven't yet gotten to that point. Do you think it should be deferred to a separate PR?

Reply:
I would advocate a separate PR, where `involutive` (or maybe `involution`) is defined for any group (or maybe even any monoid as `g ^ 2 = 1`).

Reply:
Note we already have `function.involutive e` which means the same thing as this.

Reply:
The monoid version would be `function.involutive ((*) x)`